debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -597,6 +598,34 @@ private long asEpochMicros(String timestamp) {
 
         String expectedTzLarge = "+21016-11-04T06:51:30.123456Z";
 
+        // The assertion for minimimum timestamps is problematic as it seems that Java and PostgreSQL handles conversion from large negative date
+        // to microseconds in different way
+        // written to database: 4713-12-31T23:59:59.999999 BC
+        // arrived from decoding plugin: -210831897600000001
+        // value from decoding plugin converted to Instant: -4712-12-31T23:59:59.999999Z - 1 year difference
+        // JDBC driver delivers Timestamp B.C.E. 4713-12-31T23:59:59.000+0100 that internally contains -210835184391000001 and translates to Instant -4712-11-23T22:59:59.999999Z
+        // The result is that JDBC driver and logical decoding plugin provides different values which moreover
+        // do not map to Java conversions
+        // It seems that JDBC driver always uses Gregorian calendar while Java Time API uses Julian
+        // Also it seems that further distortions could be introduced by timezone used so instead of matching
+        // against exact value the requested date should be smaller than -4712-11-01
+        final SchemaAndValueField.Condition largeNegativeTimestamp = (String fieldName, Object expectedValue, Object actualValue) -> {
+            final long expectedMinTsStreaming = -210831897600000001L;
+            final long expectedMinTsSnapshot = LocalDateTime.of(-4712, 11, 1, 0, 0, 0).toInstant(java.time.ZoneOffset.UTC).getEpochSecond() * 1_000_000;
+            final long ts = (long) actualValue;
+            assertTrue("Negative timestamp don't match for " + fieldName + ", got " + actualValue, ts >= expectedMinTsSnapshot && ts < 0);
+        };
+        // The same issue is with timezoned timestamp
+        // Database: 4714-12-31T23:59:59.999999Z BC
```

Comment:
Hum, so you persist a value long way in the future, but then what Debezium exposes will be a value long way in the past? This seems not right?

Reply:
@gunnarmorling But whose value is the correct one?